src/OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed/Configuration/EnvironmentConfigurationHelper.cs

```diff
@@ -1,20 +1,45 @@
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using OpenTelemetry.Resources;
 using OpenTelemetry.Trace;
 
 namespace OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed.Configuration
 {
     internal static class EnvironmentConfigurationHelper
     {
+        private static readonly Dictionary<IntegrationIds, Action<TracerProviderBuilder>> AddInstrumentation = new()
+        {
+            [IntegrationIds.HttpClient] = builder => builder.AddHttpClientInstrumentation(),
+            [IntegrationIds.AspNet] = builder => builder.AddSdkAspNetInstrumentation(),
+            [IntegrationIds.SqlClient] = builder => builder.AddSqlClientInstrumentation()
+        };
+
         public static TracerProviderBuilder UseEnvironmentVariables(this TracerProviderBuilder builder, Settings settings)
         {
             var resourceBuilder = ResourceBuilder
                 .CreateDefault()
                 .AddService(settings.ServiceName ?? "UNKNOWN_SERVICE_NAME", serviceVersion: settings.ServiceVersion);
 
-            return builder
+            builder
                 .SetResourceBuilder(resourceBuilder)
                 .SetExporter(settings);
+
+            foreach (var enabledInstrumentation in settings.EnabledIntegrations)
+            {
+                if (AddInstrumentation.ContainsKey(enabledInstrumentation))
```

Comment:
nitpicking since in this context this happens only at initialization but the code ends up with two lookups on the dictionary a `TryGetValue` will be better.

Also in the spirit of breaking in case of wrong configuration if `TryGetValue` returns false it should crash the app.